Provide details about block editor label syntax (#4476)
umbraco/UmbracoDocs: Fundamentals/Backoffice/Property-Editors/Built-in-Property-Editors/Block-Editor/label-property-configuration.md

---

> ---

Reviewer: Which major versions of Umbraco does this article apply to?

Author: NC was brought in in 7.7, so I believe all apply from then

# Configuring block editor label properties

When configuring a block, the label property allows you to define a label for the appearance of the Block in the editor. The label can use AngularJS template string syntax to display values of properties. Example: `My Block {{myPropertyAlias}}` will be shown as: `My Block FooBar`.

> You can also use more advanced expression using AngularJS filters, e.g. `{{myPropertyAlias | limitTo:100}}` or for a property using Richtext editor `{{myPropertyAlias | ncRichText | truncate:true:100}}`.

Reviewer: ```suggestion
You can also use more advanced expressions using AngularJS filters. Example: `{{myPropertyAlias | limitTo:100}}` or for a property using Richtext editor `{{myPropertyAlias | ncRichText | truncate:true:100}}`.
```

It is also possible to use properties from the settings model by using `{{$settings.propertyAlias}}`.

## Useful Angular filters

> As well as the [default AngularJS filters](https://docs.angularjs.org/api/ng/filter), Umbraco ships with some additional filters which are useful for setting the label field of block editors.

Reviewer: ```suggestion
As well as the [default AngularJS filters](https://docs.angularjs.org/api/ng/filter), Umbraco ships with some additional filters which are useful for setting the Label field of Block editors.
```

| [umbWordLimit](https://apidocs.umbraco.com/v10/ui/#/api/umbraco.filters.filter:umbWordLimit) | Truncates to a number of words (rather than characters) | String                      | n: maximum number of words in string                         |
| [umbCmsTitleCase](https://apidocs.umbraco.com/v10/ui/#/api/umbraco.filters.filter:umbCmsTitleCase) | Converts a string to title case                         | String                      |                                                              |
| [umbCmsJoinArray](https://apidocs.umbraco.com/v10/ui/#/api/umbraco.filters.filter:umbCmsJoinArray) | Joins an array into one string                          | Array (of string or object) | separator: string used to join values together, e.g. "`, `"<br />prop (optional): string key indicating which property to join when used on an array of objects |

### Custom filters

> If the above filters don't suit your needs, you can create custom filters by creating a plugin in `App_Plugins` and adding a filter module.

Reviewer: ```suggestion
If the filters do not suit your needs, you can create custom filters by creating a plugin in `App_Plugins` and adding a filter module.
```
